lib/puppet/functions/binary_file.rb

```diff
@@ -0,0 +1,25 @@
+# Loads a binary file from a module or file system and returns its contents as a Binary.
+# (Documented in 3.x stub)
+#
+# @since 4.8.0
+#
+Puppet::Functions.create_function(:binary_file, Puppet::Functions::InternalFunction) do
+  dispatch :binary_file do
+    scope_param
+    param 'String', :path
+  end
+
+  def binary_file(scope, unresolved_path)
+    path = nil
+    found = Puppet::Parser::Files.find_file(unresolved_path, scope.compiler.environment)
+    if found && Puppet::FileSystem.exist?(found)
```

Comment:
I see now that if an absolute path is used, the `#exists?` call is still needed since `#find_file` simply assumes that the absolute path exists. Guess that's another thing we could take a look at if/when we do a bigger overhaul to lessen the number of file accesses.